services/dummy-test-service/main.go

```diff
@@ -1,7 +1,271 @@
+			log.Fatalf("Error happened in Atoi Err: %s", err2)
+		}
+
+		jsonResp, err := json.Marshal(s.db.getBuilding(url[2], url3, url4))
+		if err != nil {
+			log.Fatalf("Error happened in JSON marshal. Err: %s", err)
+		}
+		w.Write(jsonResp)
+	} else {
+		io.WriteString(w, "Looks like you are trying to access an inexistent endpoint.\n")
+	}
+}
+
+func URLValidation(url []string, mapdb map[string][]user, w http.ResponseWriter) int {
+
+	flag := 0
+
+	users := url[1]
+
+	switch {
+	case len(url) == 4:
+		{
+			if users != "users" {
+				io.WriteString(w, "Looks like you are trying to access an inexistent endpoint.\n")
+			} else {
+
+				for i := 0; i < len(mapdb["Users"]); i++ {
+					if url[2] == mapdb["Users"][i].Username {
+						flag = 1
+						break
+					}
+				}
+			}
+		}
+	case len(url) == 5:
+		{
+			if users != "users" {
+				io.WriteString(w, "Looks like you are trying to access an inexistent endpoint.\n")
+				log.Fatalf("Error! Trying to access an inexistent endpoint.\n")
+			} else {
+				url3, err1 := strconv.Atoi(url[3])
+				if err1 != nil {
+					log.Fatalf("Error happened in Atoi Err: %s", err1)
+				}
+
+				for i := 0; i < len(mapdb["Users"]); i++ {
+					if url[2] == mapdb["Users"][i].Username {
+						for j := 0; j < len(mapdb["Users"][i].City); j++ {
+							if url3 == mapdb["Users"][i].City[j].CityID {
+								flag = 1
+								break
+							}
+						}
+					}
+					if flag == 1 {
+						break
+					}
+				}
+			}
+		}
+	case len(url) == 6:
+		{
+			if users != "users" {
+				io.WriteString(w, "Looks like you are trying to access an inexistent endpoint.\n")
+				log.Fatalf("Error! Trying to access an inexistent endpoint.\n")
+			} else {
+				url3, err1 := strconv.Atoi(url[3])
+				if err1 != nil {
+					log.Fatalf("Error happened in Atoi Err: %s", err1)
+				}
+
+				url4, err2 := strconv.Atoi(url[4])
+				if err2 != nil {
+					log.Fatalf("Error happened in Atoi Err: %s", err2)
+				}
+
+				for i := 0; i < len(mapdb["Users"]); i++ {
+					if url[2] == mapdb["Users"][i].Username {
+						for j := 0; j < len(mapdb["Users"][i].City); j++ {
+							if url3 == mapdb["Users"][i].City[j].CityID {
+								for k := 0; k < len(mapdb["Users"][i].City[j].Buildings); k++ {
+									if url4 == mapdb["Users"][i].City[j].Buildings[k].BuildingID {
+										flag = 1
+										break
+									}
+								}
+							}
+							if flag == 1 {
+								break
+							}
+						}
+					}
+					if flag == 1 {
+						break
+					}
+				}
+			}
+		}
+	default:
+		break
+	}
+	return flag
+}
+
+func (s *serverHandler) init_db() mockDB {
+
+	var database mockDB
+	database.dbmock = s.db.getUserList()
+
+	return database
+}
+
+func (s *serverHandler) handleRoot(w http.ResponseWriter, r *http.Request) {
+
+	io.WriteString(w, "Welcome to Stickian!\n")
+}
 
 func main() {
-	fmt.Printf("hello world")
+
+	server := serverHandler{
+		db: &mockDB{},
+	}
+
+	server.init_db()
+
+	client, ctx, cancel, err1 := connect("mongodb://localhost:27017")
```

Comment:
First `err1` naming does not make sense here, since you check right after for its value, you can later overwrite it.
If the mongo DB is not yet fully implemented in this PR it should also not come in. But let's let that pass...

Then the lifecycle of the process, which is usually controlled by the context, should be derived from main. So at least do something like this (and only for the data base calls should you use the db context) :
```suggestion
        // ensure external SIGINT and SIGTERM are gracefully handled
        ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
	defer stop()
	
	client, dbcontext, dbcancel, err := connect(ctx, "mongodb://localhost:27017")
```

Reply:
oops forgot to change the error variable name. Also I'm kinda new to MongoDB so please don't be too harsh :c